### PR TITLE
Email component 

### LINKED ISSUE #132: User is required to sign in before searching for parks

### PR DESCRIPTION
**Pull request by nicolegooden**
### Participating Group Members:

- @nicolegooden 

### Code Highlights:

- Prompts user to sign in with an email when app mounts
- Includes error handling for email input that is empty, doesn't include `@`, or doesn't include `.` (to be expanded at a later date)
- Hides `FindParks` unless an email is stored 
- Adds relevant state and mutations to the `Vuex store`
- Updates test in `MyParks`
- Prompts user to sign in with email if visiting `MyParks` view prior to signing in from home

### Have Tests Been Added?

- [X] No.
- [ ] Yes, but all tests are not passing.
- [ ] Yes, and all are passing.

### Any background context you want to provide?

I have skipped a test in `MyParks` and have not added any new tests for this functionality given the time constraint.  I am planning to add more test suites and update and breaking tests in the near future after our code freeze has passed.

### Message/Questions for reviewer:

N/A

### Issues:

- Closes #132 

### Screenshots (if appropriate):

Screen Shot 2021-01-13 at 12 32 34 PM
Screen Shot 2021-01-13 at 12 32 59 PM

### Tracking Consistency:

- [X] added appropriate labels
- [X] My code follows the code style of this project and has removed all unnecessary annotations
- [ ] I have added comments on my pull request, particularly in hard-to-understand areas
- [ ] All new and existing tests passed

- [X] looked at PR preview to check spelling, syntax, formatting, and completion

### CODE REVIEW

File: src/components/Email.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  methods: {
+    async submitEmail() {
+      if (this.isInvalid) {
+        this.message = 'Please enter a valid email address to continue. Required: @ Required: . '
+        this.email = ''
+        return 
+      }
+      await this.$store.commit('storeEmail', this.email);
+      this.message = `Success! ${this.$store.state.email} has been saved.`
+      setTimeout(() => this.$store.commit('changeToSearch'), 2500);
```
Comment (npdarrington) [inline src/components/Email.vue:42]: Really appreciate the error messages for a user! Great consumer empathy.

File: src/components/Email.vue [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  },
+  computed: {
+    isInvalid() {
+      return this.email === '' || !this.email.includes('@') || !this.email.includes('.')
+    }
```
Comment (npdarrington) [inline src/components/Email.vue:48]: Nice error checking!

File: src/store/index.js [hunk 3]
```diff
@@ -18,6 +20,15 @@ export default new Vuex.Store({
...
       state.geolocation = geolocation
+    },
+    storeEmail (state, email) {
+      state.email = email
+    },
+    clearResults (state) {
+      state.searchResults = []
+    },
+    changeToSearch (state) {
+      state.homeChildComp = 'FindPark'
```
Comment (npdarrington) [inline src/store/index.js:31]: Glad we were able to control it this way. We can revisit this later and refine it.

**Review by npdarrington — APPROVED** [review]
Thank you for the hard work on this Nicole!

File: src/store/index.js (hunk 3 above)
Comment (nicolegooden) [reply src/store/index.js:31]: Yeah, this doesn't seem like data that _needs_ to exist in the store.  We'll have to consider a different way to manage the data locally that allows the user to search without re-entering the email address each time that view is visited.

File: src/store/index.js [hunk 4]
```diff
@@ -18,6 +20,15 @@ export default new Vuex.Store({
     },
     updateGeolocation (state, geolocation) {
       state.geolocation = geolocation
+    },
+    storeEmail (state, email) {
```
Comment (bretm9) [inline src/store/index.js:24]: Thanks for adding these mutations. That gives us flexibility if we ever want to do this functionality elsewhere

**Review by bretm9 — APPROVED** [review]
All looks great to me! Using email as a unique identifier will make for a much better individual user experience with saving parks.

**Review by stacyp2006 — APPROVED** [review]
Nice work on this, Nicole!

**Pull request merged**